### ISSUE TITLE
Unable to play pcap & missing test folder...

### ISSUE THREAD

**Issue by mertsarica** [ctxis/RDP-Replay#6]
Hello,

I am having difficulties playing pcap file with RDP Replay tool on Ubuntu 14.04.
I extracted certificate with mimikatz from Windows 7 and also extracted LSA keys with extractrdpkeys.x86.exe.
Neither -p x509.pem nor -L HYDRAENC key works.  It says;

./rdp_replay  -r ../../rdp6.pcap  -L ../../L\$HYDRAENCKEY_28ada6da-d622-11d1-9cb9-00c04fb16e75.bin Processed private key from L$HYDRAENCKEY_28ada6da-d622-11d1-9cb9-00c04fb16e75.bin
RDP SSL MODE Requested by server!!
SSL-ERROR: No matching private key found
 
./rdp_replay  -r ../../rdp3.pcap -p ../../outfile.pem  --no_cksum
RDP SSL MODE Requested by server!!
SSL private key found.
SSL-ERROR: RSA private key decrypt failed

I am not sure what is wrong with it and in order to verify my steps, could you share your demo1.pem and demo1.pcap please because Test folder does not exist in your folder.

Regards,

**Comment by SteveWare** [comment]
This should be done. There is a README in the test directory, but this is what you are interested in:
 ```rdp_replay -r demo1.pcap -p demo1.pem --no_cksum```
I suspect that you do not have the correct private key. Good luck.

**Comment by mertsarica** [comment]
Well let me explain you the steps that I followed one by one;

In Windows 7, I ran cmd.exe as an Administrator and then ran mimikatz.exe  with psexec -s parameter with folowing commands as shown below.
C:\Users\IEUser\Desktop\Win32>PsExec.exe -s C:\Users\IEUser\Desktop\win32\mimika
tz.exe

PsExec v2.11 - Execute processes remotely
Copyright (C) 2001-2014 Mark Russinovich
Sysinternals - www.sysinternals.com


mimikatz 1.0 x86 (RC)   /* Traitement du Kiwi (Aug 26 2012 12:48:16) */
// http://blog.gentilkiwi.com/mimikatz

mimikatz # privilege::debug
Demande d'ACTIVATION du privil├¿ge : SeDebugPrivilege : OK

mimikatz # crypto::patchcapi
Patterns CRYPT_EXPORTABLE | CRYPT_ARCHIVABLE et CRYPT_ARCHIVABLE trouv├⌐s !
Patch CRYPT_EXPORTABLE | CRYPT_ARCHIVABLE : OK
Patch CRYPT_ARCHIVABLE                    : OK

mimikatz # crypto::patchcng
Service : CNG Key Isolation
Recherche des patterns dans : ncrypt.dll@pid(476)
Patch ncrypt.dll@pid(476) : OK

mimikatz # crypto::exportCertificates CERT_SYSTEM_STORE_LOCAL_MACHINE "Remote De
sktop"
Emplacement : 'CERT_SYSTEM_STORE_LOCAL_MACHINE'\Remote Desktop
         - IE10Win7
                Container Cl├⌐ : TSSecKeySet1
                Provider      : Microsoft Strong Cryptographic Provider
                Type          : AT_KEYEXCHANGE
                Exportabilit├⌐ : NON
                Taille cl├⌐    : 2048
                Export priv├⌐ dans  'CERT_SYSTEM_STORE_LOCAL_MACHINE_Remote Desk
top_0_IE10Win7.pfx' : OK
                Export public dans 'CERT_SYSTEM_STORE_LOCAL_MACHINE_Remote Deskt
op_0_IE10Win7.der' : OK

Then I installed OpenSSL-Win32 on Windows 7 and then converted pfx to pem.

C:\OpenSSL-Win32\bin>openssl pkcs12 -in "CERT_SYSTEM_STORE_LOCAL_MACHINE_Remote
Desktop_0_IE10Win7.pfx" -nodes -out x509.pem
WARNING: can't open config file: /usr/local/ssl/openssl.cnf
Enter Import Password:
MAC verified OK

Then I installed Wireshark to Windows 7, sniffed the traffic with filter "tcp.port == 3389" and then connected to that Windows 7 from Windows 8.1 via RDP (mstsc).

Then I copied sniffed traffic (Wireshark - save as - Wireshark/tcpdump - rdp.pcap) to Ubuntu 14.04 (which I successfully played your demo1.pcap) with the x509.pem of Windows 7 and then tried to play with rdp_replay.

root@ubuntu:~/Desktop/RDP-Replay-master/replay# ./rdp_replay -r rdp.pcap -p x509.pem --no_cksum
root@ubuntu:~/Desktop/RDP-Replay-master/replay# 

It shows nothing. So any idea which step is wrong ? If you'd like to get pcap, pem file and pfx, I can send it to you.

Regards,
